superset/charts/commands/data.py

```diff
@@ -0,0 +1,101 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+import logging
+from typing import Any, Dict
+
+from flask import Request
+from marshmallow import ValidationError
+
+from superset import cache
+from superset.charts.commands.exceptions import (
+    ChartDataCacheLoadError,
+    ChartDataQueryFailedError,
+)
+from superset.charts.schemas import ChartDataQueryContextSchema
+from superset.commands.base import BaseCommand
+from superset.common.query_context import QueryContext
+from superset.exceptions import CacheLoadError
+from superset.extensions import async_query_manager
+from superset.tasks.async_queries import load_chart_data_into_cache
+
+logger = logging.getLogger(__name__)
+
+
+class ChartDataCommand(BaseCommand):
```

Comment:
Primary command module for running chart queries (sync or async) via new `chart/data` API.